asf_heat_pump_suitability/analysis/exploratory/create_full_dataset/create_full_dataset_plymouth.py

```diff
@@ -0,0 +1,1182 @@
+plymouth_residential_uprns_gdf = os_uprn_plymouth_gdf[
+    (~os_uprn_plymouth_gdf["UPRN"].isin(exclude_uprns))
+    & (os_uprn_plymouth_gdf["UPRN"].isin(uprns_in_buildings))
+]
+
+# Filter building footprints to only those which contain residential UPRNs
+plymouth_residential_buildings_gdf = os_openmap_buildings_plymouth_gdf.sjoin(
+    plymouth_residential_uprns_gdf, how="inner", predicate="contains"
+).drop_duplicates(subset=["ID"])
+plymouth_residential_buildings_gdf["building_area_m2"] = (
+    plymouth_residential_buildings_gdf.area
+)
+
+# %%
+plymouth_residential_buildings_gdf.plot()
+
+# %%
+plymouth_residential_uprns_gdf.shape
+
+# %% [markdown]
+# ## Add features
+#
+# - in listed building
+# - in building conservation area
+# - outdoor space
+# - in existing / planned heat network zone
+# - tenure
+# - property type
+# - heating fuel
+# - off gas status
+
+# %%
+# Identify UPRNs in listed buildings
+listed_building_uprns = (
+    plymouth_residential_uprns_gdf.sjoin_nearest(
+        listed_buildings_gdf, how="inner", max_distance=10
+    )["UPRN"]
+    .unique()
+    .tolist()
+)
+
+# UPRNs in building conservation areas
+cons_area_uprns = plymouth_residential_uprns_gdf.sjoin(
+    cons_areas_gdf.to_crs(epsg=27700), how="inner", predicate="intersects"
+)["UPRN"].tolist()
+
+# Reformat UPRN for join
+outdoor_space_df = outdoor_space_df.with_columns(
+    pl.col("UPRN").str.replace(".0", "", literal=True).cast(pl.Int64).alias("UPRN")
+)
+
+# UPRNs in HNs
+uprns_in_hn = plymouth_residential_uprns_gdf.sjoin(
+    hn_zones_gdf.drop(columns=["index_right"]), predicate="within"
+)["UPRN"].tolist()
+
+# %%
+len(listed_building_uprns) / len(plymouth_residential_uprns_gdf) * 100
+
+# %%
+# Standardise postcode column - remove space and set to upper case
+epc_df = output_areas.standardise_col_postcode(epc_df, pcd_col="POSTCODE")
+epc_df = epc_df.with_columns(
+    # Standardise UPRN
+    pl.col("UPRN")
+    .str.replace(".0", "", literal=True)
+    .cast(pl.Int64, strict=False)
+    .alias("UPRN"),
+    pl.col("MAIN_FUEL").cast(pl.String).alias("MAIN_FUEL"),
+    pl.col("PROPERTY_TYPE").cast(pl.String).alias("PROPERTY_TYPE"),
+)
+
+# Process central heating fuel type information
+epc_df = fuel_type.extend_df_fuel_type(epc_df, epc_col="MAIN_FUEL", name="fuel_type")
+epc_df = fuel_type.extend_df_fuel_type(
+    epc_df, epc_col="MAINHEAT_DESCRIPTION", name="fill_fuel_type"
+)
+epc_df = epc_df.with_columns(
+    pl.col("fuel_type").fill_null(pl.col("fill_fuel_type"))
+).drop(["fill_fuel_type"])
+
+# %%
+# Create features dataset with initial property-level features
+features_df = (
+    pl.from_pandas(plymouth_residential_uprns_gdf[["UPRN"]])
+    .with_columns(
+        pl.when(pl.col("UPRN").is_in(listed_building_uprns))
+        .then(True)
+        .otherwise(False)
+        .alias("in_listed_building"),
+        pl.when(pl.col("UPRN").is_in(cons_area_uprns))
+        .then(True)
+        .otherwise(False)
+        .alias("in_cons_area"),
```

Comment:
```suggestion
        .alias("in_conservation_area"),
```
just makes it more readable